src/beast/beast/crypto/impl/sha2_context.h

```diff
@@ -80,41 +95,192 @@ struct sha512_context
+    *((str) + 1) = (std::uint8_t) ((x) >> 16);  \
+    *((str) + 0) = (std::uint8_t) ((x) >> 24);  \
 }
 #define BEAST_SHA2_PACK64(str, x)               \
 {                                               \
-    *(x) =   ((std::uint64_t) *((str) + 7)      )      \
-           | ((std::uint64_t) *((str) + 6) <<  8)      \
-           | ((std::uint64_t) *((str) + 5) << 16)      \
-           | ((std::uint64_t) *((str) + 4) << 24)      \
-           | ((std::uint64_t) *((str) + 3) << 32)      \
-           | ((std::uint64_t) *((str) + 2) << 40)      \
-           | ((std::uint64_t) *((str) + 1) << 48)      \
-           | ((std::uint64_t) *((str) + 0) << 56);     \
+    *(x) =                                      \
+          ((std::uint64_t) *((str) + 7)      )  \
+        | ((std::uint64_t) *((str) + 6) <<  8)  \
+        | ((std::uint64_t) *((str) + 5) << 16)  \
+        | ((std::uint64_t) *((str) + 4) << 24)  \
+        | ((std::uint64_t) *((str) + 3) << 32)  \
+        | ((std::uint64_t) *((str) + 2) << 40)  \
+        | ((std::uint64_t) *((str) + 1) << 48)  \
+        | ((std::uint64_t) *((str) + 0) << 56); \
+}
+#define BEAST_SHA2_UNPACK64(x, str)             \
+{                                               \
+    *((str) + 7) = (std::uint8_t) ((x)      );  \
+    *((str) + 6) = (std::uint8_t) ((x) >>  8);  \
+    *((str) + 5) = (std::uint8_t) ((x) >> 16);  \
+    *((str) + 4) = (std::uint8_t) ((x) >> 24);  \
+    *((str) + 3) = (std::uint8_t) ((x) >> 32);  \
+    *((str) + 2) = (std::uint8_t) ((x) >> 40);  \
+    *((str) + 1) = (std::uint8_t) ((x) >> 48);  \
+    *((str) + 0) = (std::uint8_t) ((x) >> 56);  \
+}
+
+//------------------------------------------------------------------------------
+
+// SHA256
+
+template <class = void>
+void sha256_transform (sha256_context& ctx,
+    unsigned char const* message,
+        unsigned int block_nb) noexcept
+{
+    static unsigned long long const K[64] = {
+        0x428a2f98, 0x71374491, 0xb5c0fbcf, 0xe9b5dba5,
+        0x3956c25b, 0x59f111f1, 0x923f82a4, 0xab1c5ed5,
+        0xd807aa98, 0x12835b01, 0x243185be, 0x550c7dc3,
+        0x72be5d74, 0x80deb1fe, 0x9bdc06a7, 0xc19bf174,
+        0xe49b69c1, 0xefbe4786, 0x0fc19dc6, 0x240ca1cc,
+        0x2de92c6f, 0x4a7484aa, 0x5cb0a9dc, 0x76f988da,
+        0x983e5152, 0xa831c66d, 0xb00327c8, 0xbf597fc7,
+        0xc6e00bf3, 0xd5a79147, 0x06ca6351, 0x14292967,
+        0x27b70a85, 0x2e1b2138, 0x4d2c6dfc, 0x53380d13,
+        0x650a7354, 0x766a0abb, 0x81c2c92e, 0x92722c85,
+        0xa2bfe8a1, 0xa81a664b, 0xc24b8b70, 0xc76c51a3,
+        0xd192e819, 0xd6990624, 0xf40e3585, 0x106aa070,
+        0x19a4c116, 0x1e376c08, 0x2748774c, 0x34b0bcb5,
+        0x391c0cb3, 0x4ed8aa4a, 0x5b9cca4f, 0x682e6ff3,
+        0x748f82ee, 0x78a5636f, 0x84c87814, 0x8cc70208,
+        0x90befffa, 0xa4506ceb, 0xbef9a3f7, 0xc67178f2
+    };
+    std::uint32_t w[64];
+    std::uint32_t wv[8];
+    std::uint32_t t1, t2;
+    unsigned char const* sub_block;
+    int i, j;
```

Comment:
is this all copied from somewhere, and thus their weirdness?  Why not define these variables (sub_block, i, j) where they are used?